test/validator/gitignore_test.dart

```diff
@@ -51,6 +51,13 @@ void main() {
       ]),
       exit_codes.DATA,
     );
+
+    // Check that you can unignore certain files too.
+    await d.dir('myapp', [
+      d.file('.gitignore', '*.txt\n!foo.txt'),
```

Comment:
I believe this is tested well enough in the general ignore-rules tests. Eg. here: https://github.com/dart-lang/pub/blob/e1fbda73d1ac597474b82882ee0bf6ecea5df108/test/ignore_test.dart#L344